lib/internal/modules/esm/module_job.js

```diff
@@ -45,22 +48,43 @@ const isCommonJSGlobalLikeNotDefinedError = (errorMessage) =>
     (globalLike) => errorMessage === `${globalLike} is not defined`,
   );
 
-/* A ModuleJob tracks the loading of a single Module, and the ModuleJobs of
- * its dependencies, over time. */
+/**
+ * A ModuleJob tracks the loading of a single Module, and the ModuleJobs of its dependencies, over
+ * time.
+ */
 class ModuleJob {
-  // `loader` is the Loader instance used for loading dependencies.
-  // `moduleProvider` is a function
-  constructor(loader, url, importAssertions = { __proto__: null },
-              moduleProvider, isMain, inspectBrk) {
-    this.loader = loader;
+  /**
+   * Deep dependency jobs wrappers are instantiated, and module wrapper is instantiated.
+   */
+  instantiated;
+
+  module;
+
+  constructor(
+    url,
+    moduleWrapper,
+    importAssertions = { __proto__: null },
+    isMain,
+    inspectBrk,
+  ) {
     this.importAssertions = importAssertions;
     this.isMain = isMain;
     this.inspectBrk = inspectBrk;
+    this.modulePromise = moduleWrapper;
+
+    debug('new ModuleJob(%o)', { url, importAssertions, isMain, inspectBrk });
 
-    this.module = undefined;
     // Expose the promise to the ModuleWrap directly for linking below.
     // `this.module` is also filled in below.
-    this.modulePromise = ReflectApply(moduleProvider, loader, [url, isMain]);
+
+    setCallbackForWrap(this.modulePromise, {
+      initializeImportMeta: (meta, wrap) => this.importMetaInitialize(meta, { url }),
+      importModuleDynamically: (specifier, { url }, importAssertions) => {
+        const moduleLoader = require('internal/process/esm_loader').esmLoader;
+        debug('importModuleDynamically: %o', { specifier, url, moduleLoader });
+        return moduleLoader.import(specifier, url, importAssertions);
+      },
+    });
```

Comment:
I don't think it's the right place for this, IMO it makes more sense to keep it in translators.

Reply:
It think `ESMLoader::eval()` doesn't use a translator; I was trying to consolidate the logic to ensure it behaves the same.

I do agree that its current state is messy (hence the "WIP" commit though)